File: app/src/main/java/yangfentuozi/runner/service/data/EnvInfo.java
package yangfentuozi.runner.service.data;

import android.os.Parcel;
import android.os.Parcelable;

public class EnvInfo implements Parcelable {

    public String key;
    public String value;

    public EnvInfo() {
    }

    public EnvInfo(Parcel source) {
        super();
        key = source.readString();
        value = source.readString();
    }

    public int describeContents() {
        return 0;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(key);
        dest.writeString(value);
    }

    public static final Parcelable.Creator<EnvInfo> CREATOR = new Parcelable.Creator<>() {
        public EnvInfo createFromParcel(Parcel source) {
            return new EnvInfo(source);
        }

        public EnvInfo[] newArray(int size) {
            return new EnvInfo[size];
        }
    };
}

File: shared/src/main/java/yangfentuozi/runner/shared/data/EnvInfo.kt
package yangfentuozi.runner.shared.data

import android.os.Parcel
import android.os.Parcelable

open class EnvInfo : Parcelable {
    var key: String? = null
    var value: String? = null
    var enabled: Boolean = true

    constructor()

    constructor(source: Parcel) : super() {
        key = source.readString()
        value = source.readString()
        enabled = source.readBoolean()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(key)
        dest.writeString(value)
        dest.writeBoolean(enabled)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<EnvInfo?> = object : Parcelable.Creator<EnvInfo?> {
            override fun createFromParcel(source: Parcel): EnvInfo {
                return EnvInfo(source)
            }

            override fun newArray(size: Int): Array<EnvInfo?> {
                return arrayOfNulls<EnvInfo>(size)
            }
        }
    }
}